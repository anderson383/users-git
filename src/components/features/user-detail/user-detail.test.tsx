
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import UserDetail from '.';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

const mockUser = {
  name: 'Pepito xd',
  avatar_url: 'url image',
  login: 'test user',
  id: 'id-test',
  twitter_username: '',
  bio: 'biografia',
  location: 'Colombia'
}

const navigateMock = jest.fn()

const saveUserMock = jest.fn().mockImplementation(() => ('Success'))

jest.mock('react-router-dom', () => {
  return ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => jest.fn(),
    useNavigate: () => navigateMock
  }
  )
})

jest.mock('../../../services/config/context', () => ({
  useRepositoryIoc: jest.fn().mockReturnValue({
    container: {
      get: jest.fn().mockReturnValue({
        getUser: jest.fn().mockImplementation(() => Promise.resolve(mockUser)),
        saveUser: (argumnts:any) => saveUserMock(argumnts)
      })
    }
  })
}))

window.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}))

describe('Tests in Detail component', () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <UserDetail />
      </BrowserRouter>
    );
  };

  test('Should render the component', async () => {
    const {
      asFragment
    } = setup();
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should click export save user', async () => {

    setup();

    await act(() => Promise.resolve())
    const button = screen.getByText('Exportar')

    fireEvent.click(button)

    expect(saveUserMock).toHaveBeenCalledWith({
      name: mockUser.name,
      jsonData: mockUser
    });
  });
});
