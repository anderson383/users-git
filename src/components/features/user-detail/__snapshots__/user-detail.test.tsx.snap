// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Detail component Should render the component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      Detalle de usuario
    </h2>
    <div
      class="export"
    >
      <button>
        Exportar
      </button>
    </div>
    <div
      class="card_detail"
    >
      <div
        class="avatar"
      >
        <div>
          <img
            alt=""
            src="url image"
            width="300"
          />
        </div>
        <div
          class="info"
        >
          <b>
            Pepito xd
          </b>
          <p>
            Twitter: No tiene
          </p>
          <p>
            Followers: No tiene
          </p>
        </div>
      </div>
      <div
        class="info_company"
      >
        <h2>
          Informaci√≥n
        </h2>
        <p>
          <b>
            Company:
          </b>
            No tiene
        </p>
        <p>
          <b>
            Location:
          </b>
           Colombia
        </p>
        <p>
          <b>
            Biografia:
          </b>
           biografia
        </p>
        <p>
          <b>
            Email:
          </b>
           No tiene
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
